src/Cli/Microsoft.TemplateEngine.Cli/TemplatePackageCoordinator.cs

```diff
@@ -332,6 +381,198 @@ internal async Task<NewCommandStatus> EnterUninstallFlowAsync(UninstallCommandAr
+                    packageIdentity,
+                    packageVersion,
+                    sourceFeed,
+                    cancellationToken).ConfigureAwait(false);
+
+                if (nuGetPackageMetadata == null)
+                {
+                    DisplayLocalPackageMetadata(localPackage, Reporter.Output);
+
+                    var templatesToDisplay = TemplateGroupDisplay.GetTemplateGroupsForListDisplay(packageTemplates, null, null, _engineEnvironmentSettings.Environment);
+                    DisplayPackageTemplateList(templatesToDisplay, Reporter.Output);
+                    return NewCommandStatus.Success;
+                }
+            }
+            else
+            {
+                IEnumerable<PackageSource> packageSources = LoadNuGetSources(additionalSources, true);
+
+                nuGetPackageMetadata = await GetPackageMetadataFromMultipleFeedsAsync(packageSources, nugetApiManager, packageIdentity, packageVersion, cancellationToken).ConfigureAwait(false);
+                if (nuGetPackageMetadata != null && nuGetPackageMetadata.Source.Source.Contains("api.nuget.org"))
+                {
+                    packageTemplates = await CliTemplateSearchCoordinator.SearchForPackageTemplatesAsync(
+                        _engineEnvironmentSettings,
+                        packageIdentity,
+                        packageVersion,
+                        cancellationToken).ConfigureAwait(false);
+                }
+
+            }
+
+            if (nuGetPackageMetadata != null)
+            {
+                DisplayNuGetPackageMetadata(nuGetPackageMetadata, Reporter.Output);
+                if (packageTemplates != null && packageTemplates.Any())
+                {
+                    var templatesToDisplay = TemplateGroupDisplay.GetTemplateGroupsForListDisplay(packageTemplates, null, null, _engineEnvironmentSettings.Environment);
+                    DisplayPackageTemplateList(templatesToDisplay, Reporter.Output);
+                }
+                return NewCommandStatus.Success;
+            }
+
+            Reporter.Output.WriteLine(
+                LocalizableStrings.Generic_Info_NoMatchingTemplatePackage.Bold().Red(),
+                $"{packageIdentity}{(string.IsNullOrWhiteSpace(packageVersion) ? string.Empty : $"::{packageVersion}")}");
+
+            return NewCommandStatus.NotFound;
+        }
+
+        internal void DisplayNuGetPackageMetadata(NugetPackageMetadata packageMetadata, IReporter reporter)
+        {
+            reporter.WriteLine($"{packageMetadata.Identity.Id}");
+            WriteIfNotNull(LocalizableStrings.DetailsCommand_Property_Version, packageMetadata.PackageVersion.ToString(), reporter, 1);
+
+            WriteIfNotNull(LocalizableStrings.DetailsCommand_Property_Description, packageMetadata.Description, reporter, 1);
+            WriteIfNotNull(LocalizableStrings.DetailsCommand_Property_SourceFeed, packageMetadata.Source.ToString(), reporter, 1);
```

Comment:
I don't think we should rely on the `ToString()` member of PackageSource here - it could change. We should be explicit and write the source in the following fashion:

* if source.Name == source.Source, only write the source.Source
* otherwise, write `{source.Name} [{source.Source}]`

This reduces duplication, and insulates us from tostring changes coming from NuGet APIs.